common-features/data-binding/descriptors.md

```diff
@@ -0,0 +1,317 @@
+
+This article explains how to retrieve the applied filtering, searching, sorting, and grouping criteria in Blazor components. The article applies to components that support these features. The components that offer one or all of the functionalities are:
+* The [Filter]({%slug filter-overview%})
+* The [Gantt]({%slug gantt-overview%})
+* The [TreeList]({%slug treelist-overview%})
+* All components that [expose the `OnRead` event]({%slug common-features-data-binding-onread%}#components-with-onread-event), excluding the [ListView]({%slug listview-overview%}), because the ListView doesn't support built-in filtering, searching, sorting, and grouping.
+
+## Obtain Filtering, Searching, Sorting, Grouping criteria
+
+There are two ways to obtain the applied filtering, searching, sorting, grouping criteria:
+
+#### Through the OnRead Event
+
+Use the [`Request` property]({%slug common-features-data-binding-onread%}#event-argument) of the [`OnRead` event argument object](/blazor-ui/api/Telerik.Blazor.Components.ReadEventArgs):
+
+````CS
+async Task OnReadHandler(...ReadEventArgs args)
+{
+    // Get the applied filtering and searching criteria
+    // args.Request.Filters
+
+    // Get the applied grouping criteria, including:
+    // *the field by which the user groups
+    // *the sort direction of the groups ordering
+    // args.Request.Groups
+
+    // Get the applied sorting criteria, including:
+    // *the field which the user sorts
+    // *the sort direction
+    // args.Request.Sorts
+}
+````
+
+#### Through the Component State
+Use the component state property of the `OnStateChanged` event argument. For example:
+
+````CS
+async Task OnStateChangedHandler(GridStateEventArgs<Product> args)
+{
+    // Get the applied filtering criteria
+    // args.GridState.FilterDescriptors
+
+    // Get the applied searching criteria
+    // args.GridState.SearchFilter
+
+    // Get the applied grouping criteria, including:
+    // *the field by which the user groups
+    // *the sort direction of the groups ordering
+    // args.GridState.GroupDescriptors
+
+    // Get the applied sorting criteria, including:
+    // *the field which the user sorts
+    // *the sort direction
+    // args.GridState.SortDescriptors
+}
+````
+
+At the bottom of the article you will find full examples.
+
+
+## Filtering
+
+The filtering criteria for each filtered field is stored in an individual collection of [`IFilterDescriptor`](/blazor-ui/api/Telerik.DataSource.IFilterDescriptor). To access the filtering criteria, such as the user input to filter by, cast each `IFilterDescriptor`:
+
+* If the component is of type input or select, such as the AutoComplete, the ComboBox, the DropDownList, the MultiColumnComboBox, the MultiSelect, cast each `IFilterDescriptor` to [`FilterDescriptor`](/blazor-ui/api/telerik.datasource.filterdescriptor).
+* Otherwise, cast each `IFilterDescriptor` to [`CompositeFilterDescriptor`](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor).
```

Comment:
This can be misleading, because it should apply only to the parent descriptors, but not to the children.